tabular/src/autogluon/tabular/models/image_prediction/image_predictor.py

```diff
@@ -0,0 +1,207 @@
+        if self._internal_feature_map:
+            X = X.rename(columns=self._internal_feature_map)
+        return X
+
+    def _fit(self,
+             X: pd.DataFrame,
+             y: pd.Series,
+             X_val: Optional[pd.DataFrame] = None,
+             y_val: Optional[pd.Series] = None,
+             time_limit: Optional[int] = None,
+             sample_weight=None,
+             verbosity=2,
+             **kwargs):
+        """The internal fit function
+
+        Parameters
+        ----------
+        X
+            Features of the training dataset
+        y
+            Labels of the training dataset
+        X_val
+            Features of the validation dataset
+        y_val
+            Labels of the validation dataset
+        time_limit
+            The time limit for the fit function
+        kwargs
+            Other keyword arguments
+
+        """
+        try_import_mxnet()
+        try_import_autogluon_vision()
+        from autogluon.vision import ImagePredictor, ImageDataset
+        params = self._get_model_params()
+
+        X = self.preprocess(X, fit=True)
+        if X_val is not None:
+            X_val = self.preprocess(X_val)
+
+        if sample_weight is not None:  # TODO: support
+            logger.log(15, "sample_weight not yet supported for ImagePredictorModel, this model will ignore them in training.")
+
+        X = X.reset_index(drop=True)
+        y = y.reset_index(drop=True)
+        if X_val is not None:
+            X_val = X_val.reset_index(drop=True)
+            y_val = y_val.reset_index(drop=True)
+        X[self._label_column_name] = y
+        if X_val is not None:
+            X_val[self._label_column_name] = y_val
+
+        verbosity_image = max(0, verbosity - 1)
+        root_logger = logging.getLogger()
+        root_log_level = root_logger.level
+        # TODO: ImagePredictor doesn't use problem_type in any way at present.
+        #  It also doesn't error or warn if problem_type is not one it expects.
+        self.model = ImagePredictor(
+            problem_type=self.problem_type,
+            path=self.path,
+            # eval_metric=self.eval_metric,  # TODO: Vision only works with accuracy
+            verbosity=verbosity_image
+        )
+        classes = list(y.unique())
+        classes.sort()
+        self._classes = classes
+
+        X = ImageDataset(X, classes=classes)
+        if X_val is not None:
+            X_val = ImageDataset(X_val, classes=classes)
+
+        self.model.fit(train_data=X,
+                       tuning_data=X_val,
+                       time_limit=time_limit,
+                       hyperparameters=params,
+                       random_state=0)
+        # self.model.set_verbosity(verbosity)  # TODO: How to set verbosity of fit predictor?
+        root_logger.setLevel(root_log_level)  # Reset log level
+
+    def _predict_proba(self, X, **kwargs):
+        X = self.preprocess(X, **kwargs)
+
+        if self.problem_type == REGRESSION:
```

Comment:
Should this error be moved earlier as well for `fit()`?

Reply:
added